agents-for-bedrock/use-case-examples/product-review-agent/README.md

```diff
@@ -0,0 +1,70 @@
+# Custom integration with knowledge base
+
+The agent integrates with the knowledge base without an explicit association to it. Essentially, based on the function invoked in the function definitions, it can choose to query the vector DB directly by filtering or the knowledge base through both filtering and semantic similarity. This is useful when you want to leverage on the managed knowledge base for data maintenance (create,delete,update of documents in vector database), and exert more control over invocations of the knowledge base.
+
+## Dataset
+
+This example uses the [amazon reviews 2018 dataset](https://jmcauley.ucsd.edu/data/amazon_v2/categoryFilesSmall/All_Beauty_5.json.gz) with these fields - review (string), rating (number), timestamp (number), reviewers (string list).
+
+We create the following metadata file for each of the text chunks (review) in the knowledge base.
+
+The timestamp is in epoch format so date filters using ```greaterThan/lessThan``` operators can be used since Bedrock KB doesn't support date type currently.
+
+In addition, string list can be filtered using the ```in``` operator. This can be used to implement document level access controls in OpenSearch Serverless. Each document can have a role attribute containing the list of roles that can access the document.
+
+```json
+{"metadataAttributes": {"rating": 4.0, "timestamp": 1588615855070, "reviewers": ["lebron", "jokic", "curry"]}}
+```
+
+## Knowledge Base integration
```

Comment:
There should be a more clear explanation that this repo is not using the KB native integration, but an action group that uses KB APIs. could you add a paragraph?